File: app/src/main/java/com/cartenz/cartenztaxcore/api/repository/ForgotPassRepository.java
package com.cartenz.cartenztaxcore.api.repository;

import com.cartenz.cartenztaxcore.api.BaseApiClient;
import com.cartenz.cartenztaxcore.api.dao.SimpleStringDao;
import com.google.gson.JsonObject;

import okhttp3.MediaType;
import okhttp3.RequestBody;
import rx.Observable;

/**
 * Created by pratama on 2/22/2018.
 */

public class ForgotPassRepository {
    private String email;
    private BaseApiClient apiClient;

    public ForgotPassRepository(BaseApiClient apiClient, String email) {
        this.apiClient = apiClient;
        this.email = email;
    }

    public Observable<SimpleStringDao> post() {
        JsonObject jsonObject = new JsonObject();
        jsonObject.addProperty("93010006", email);
        return apiClient.getInterfaceClass().forgotpass(RequestBody.create(MediaType.parse("application/json"), jsonObject.toString()));
    }
}

File: kotlinapp/src/main/java/com/cartenz/kotlinapp/api/repository/ForgotPassRepository.kt
package com.cartenz.kotlinapp.api.repository

import com.cartenz.kotlinapp.App
import com.cartenz.kotlinapp.api.dao.SimpleStringDao
import com.google.gson.JsonObject
import okhttp3.MediaType
import okhttp3.RequestBody
import rx.Observable

class ForgotPassRepository(private val email: String) {

    fun post(): Observable<SimpleStringDao> {
        val jsonObject = JsonObject()
        jsonObject.addProperty("93010006", email)
        return App.getApiInterface().forgotpass(RequestBody.create(MediaType.parse("application/json"), jsonObject.toString()))
    }
}
